docusaurus/docs/Android/04-compose/07-guides/02-customizing-image-and-video-previews.mdx

```diff
@@ -0,0 +1,166 @@
+    ) {
+        Icon(
+            modifier = Modifier
+                .padding(2.dp)
+                .fillMaxSize(0.8f),
+            painter = painterResource(id = R.drawable.stream_compose_ic_play),
+            tint = Color.White,
+            contentDescription = null
+        )
+    }
+}
+```
+
+Now that we have created our custom button, we can use the slots provided by `MediaAttachmentContent` to override the default play button inside both the message list and the message composer.
+We'll also use the opportunity to increase the maximum number of tiles that the factory can display.
+
+```kotlin
+val customMediaAttachmentFactory = MediaAttachmentFactory(
+    // Increase the maximum number of previewed items to 5
+    maximumNumberOfPreviewedItems = 5,
+    // Render a custom item above attachments inside the message list
+    itemOverlayContent = { attachmentType ->
+        // Apply it only to video attachments
+        if (attachmentType == AttachmentType.VIDEO) {
+            CustomPlayButton(
+                modifier = Modifier
+                    .widthIn(10.dp)
+                    .padding(2.dp)
+                    .background(
+                        color = Color(red = 256, blue = 256, green = 256, alpha = 220),
+                        shape = RoundedCornerShape(8.dp)
+                    )
+                    .fillMaxWidth(0.3f)
+                    .aspectRatio(1.20f),
+            )
+        }
+    },
+    // Render a custom item above attachments inside the message composer
+    previewItemOverlayContent = { attachmentType ->
+        // Apply it only to video attachments
+        if (attachmentType == AttachmentType.VIDEO) {
+            CustomPlayButton(
+                modifier = Modifier
+                    .padding(2.dp)
+                    .background(
+                        color = Color(red = 256, blue = 256, green = 256, alpha = 220),
+                        shape = RoundedCornerShape(8.dp)
+                    )
+                    .fillMaxWidth(0.35f)
+                    .aspectRatio(1.20f),
+            )
+        }
+    })
+```
+
+Since we only want to modify `MediaAttachmentFactory` and keep the rest of the default factories, we'll copy them in the same order they are created in [StreamAttachmentFactories.defaultFactories()](https://github.com/GetStream/stream-chat-android/blob/main/stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/StreamAttachmentFactories.kt#L65).
+
+```kotlin
+val attachmentFactories = listOf(
+    UploadAttachmentFactory(),
+    LinkAttachmentFactory(linkDescriptionMaxLines = 5),
+    GiphyAttachmentFactory(),
+    customMediaAttachmentFactory,
+    FileAttachmentFactory(),
+    UnsupportedAttachmentFactory()
+)
+```
+
+Remember, order of precedence matters when creating a list of attachment factories, as the first one whose `canHandle()` function evaluates to true will render the given attachment.
```

Comment:
```suggestion
Remember, ordering matters when creating a list of attachment factories. The first factory that returns true on `canHandle()` will be used to render the given attachment.
```

Too verbose, can be simplified.

Reply:
The original one is only 1 word and 11 characters longer than the suggestion, so I'll keep it as it's grammatically more correct.

Reply:
Even though it is grammatically correct i find it too convoluted and hard to read.